Source/WebCore/html/InputType.cpp

```diff
@@ -91,7 +91,14 @@ using namespace HTMLNames;
 typedef bool (Settings::*InputTypeConditionalFunction)() const;
 typedef const AtomString& (*InputTypeNameFunction)();
 typedef Ref<InputType> (*InputTypeFactoryFunction)(HTMLInputElement&);
-typedef MemoryCompactLookupOnlyRobinHoodHashMap<AtomString, std::pair<InputTypeConditionalFunction, InputTypeFactoryFunction>> InputTypeFactoryMap;
+
+struct InputTypeFactory {
+    AtomString type;
+    InputTypeConditionalFunction conditionalFunction;
+    InputTypeFactoryFunction factoryFunction;
+};
+
+typedef MemoryCompactLookupOnlyRobinHoodHashMap<AtomString, InputTypeFactory> InputTypeFactoryMap;
```

Comment:
I don't think we need to store the type `AtomString` in both the key and the value. Can we use `MemoryCompactLookupOnlyRobinHoodHashTable<InputTypeFactor>` instead to avoid this? Or, change `findFactory` to return the `MemoryCompactLookupOnlyRobinHoodHashMap::iterator` (or return a pair of the `AtomString` and `InputTypeFactory` pointers?) so that its caller can look at both the key and value?

Reply:
ok, I'd make this function return std::pair<const AtomString&, InputTypeFactory*>.